src/Propel/Generator/Reverse/MysqlSchemaParser.php

```diff
@@ -410,30 +410,22 @@ protected function addForeignKeys(Table $table)
                     $fcols[] = trim($piece, '` ');
                 }
 
-                // typical for mysql is RESTRICT
                 $fkactions = [
-                    'ON DELETE' => ForeignKey::RESTRICT,
-                    'ON UPDATE' => ForeignKey::RESTRICT,
+                    'ON DELETE' => null,
+                    'ON UPDATE' => null,
                 ];
 
                 if ($fkey) {
-                    // split foreign key information -> search for ON DELETE and afterwords for ON UPDATE action
+                    // split foreign key information -> search for ON DELETE and afterwards for ON UPDATE action
                     foreach (array_keys($fkactions) as $fkaction) {
                         $result = null;
-                        preg_match('/' . $fkaction . ' (' . ForeignKey::CASCADE . '|' . ForeignKey::SETNULL . ')/', $fkey, $result);
+                        preg_match('/' . $fkaction . ' (' . ForeignKey::CASCADE . '|' . ForeignKey::SETNULL . '|' . ForeignKey::RESTRICT . '|' . ForeignKey::NOACTION . ')/', $fkey, $result);
```

Comment:
Might as well clean it up while you are working on it:
```php
$availableActions = [ForeignKey::CASCADE, ForeignKey::SETNULL, ForeignKey::RESTRICT, ForeignKey::NOACTION];
$pipedActionsString = implode('|', $availableActions);
$regex = sprintf('/ %s (%s)/',$fkaction,  $pipedActionsString);
preg_match($regex, $fkey, $result);
```
Personally, I would put this into its own method `getForeignKeyBehaviors()`, but this might be asking too much.

Reply:
Done